docs(router): add guide for nested redirect routes (#552)
analogjs/analog: apps/docs-app/docs/features/routing/metadata.md

> Routes can be defined for only the purpose of redirecting to another route.

Reviewer: ```suggestion
Routes can be defined for the sole purpose of redirecting to another route.
```

};
```

> The example above is a redirect to the `/home` route from `/` route.

Reviewer: ```suggestion
The example above is a redirect from the `/` route to the `/home` route.
```

:::tip

Redirect route files should not export a component.

:::

It's also possible to define nested redirect routes. For the following file structure:

```treeview
src/
└── app/
    └── pages/
        └── cities/
            ├── index.page.ts
            ├── new-york.page.ts
            └── san-francisco.page.ts
```

> and the `routeMeta` definition in the `src/app/pages/cities/index.page.ts` file:

Reviewer: ```suggestion
and the following `routeMeta` definition to the `src/app/pages/cities/index.page.ts` file:
```

```ts
import { RouteMeta } from '@analogjs/router';

export const routeMeta: RouteMeta = {
  redirectTo: '/cities/new-york',
  pathMatch: 'full',
};
```

> navigating to `/cities` will redirect to `/cities/new-york`.

Reviewer: ```suggestion
Now, navigating to `/cities` will redirect to `/cities/new-york`. 
One final note: Nested redirects always require an absolute path.
```

};
```

The example above is a redirect from the `/` route to the `/home` route.

:::tip

Redirect route files should not export a component.

:::

It's also possible to define nested redirect routes. For the following file structure:

```treeview
src/
└── app/
    └── pages/
        └── cities/
            ├── index.page.ts
            ├── new-york.page.ts
            └── san-francisco.page.ts
```

and the following `routeMeta` definition to the `src/app/pages/cities/index.page.ts` file:

```ts
import { RouteMeta } from '@analogjs/router';

export const routeMeta: RouteMeta = {
  redirectTo: '/cities/new-york',
  pathMatch: 'full',
};
```

> navigating to `/cities` will redirect to `/cities/new-york`.

Reviewer 2: ```suggestion
Navigating to `/cities` redirects to `/cities/new-york`.
```
